actions/ci-beholder-validator/action.yml

```diff
@@ -0,0 +1,93 @@
+name: "Schema Validator Action"
+description: "Validates schema compatibility with schema registry"
+
+inputs:
+  docker-registry:
+    description: "Docker registry to use"
+    required: true
+    default: "aws"
+  image-tag:
+    description: "Image tag to use"
+    required: false
+    default: "latest"
+  # aws inputs
+  aws-role-duration-seconds:
+    description: ""
+    required: false
+    default: "900"
+  aws-region:
+    description: ""
+    required: false
+  aws-account-number:
+    description: ""
+    required: false
+  aws-role-arn:
+    description: ""
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout repo
+      uses: actions/checkout@v4.2.1
+      with:
+        fetch-depth: ${{ inputs.checkout-repo-fetch-depth }}
+    - name: Detect changed files
+      id: changes
+      shell: bash
+      run:
+        echo "CHANGED_FILES=$(git diff --name-only origin/master)" >>
+        $GITHUB_ENV
+    - name: Start Redpanda with Docker Compose
+      shell: bash
+      run: |
+        docker compose up -d redpanda-console
+    - name: Configure aws creds
```

Comment:
@erikburt I would love to have your review here. I am not that familiar with this. Will this approach to pulling images work in other repos where these actions would be used?
Are there some changes/alternatives I should explore?

Reply:
Yes this should work.

The callers of this action will need to pass an AWS role they can assume, and that role requires read-access to the ECR.